tamr_toolbox/project/_common/movement.py

```diff
@@ -0,0 +1,137 @@
+        operation = operation.wait()
+        utils.operation.enforce_success(operation)
+        logging.info("Export completed successfully.")
+
+    return operation
+
+
+def import_project(
+    *,
+    tamr_client: Client,
+    project_artifact_path: str,
+    target_project: Project = None,
+    new_project_name: str = None,
+    new_unified_dataset_name: Optional[str] = None,
+    exclude_artifacts: Optional[List[str]] = [],
+    include_additive_artifacts: Optional[List[str]] = [],
+    include_destructive_artifacts: Optional[List[str]] = [],
+    fail_if_not_present: bool = False,
+    asynchronous: bool = False,
+    overwrite_existing: bool = False,
+) -> Operation:
+    """Import project artifacts into a tamr instance
+
+    Version:
+        Requires Tamr 2021.005.0 or later
+
+    Args:
+        tamr_client: a tamr client
+        project_artifact_path: project artifacts zip filepath
+        target_project: an optional target project for migration
+        new_project_name: new project name
+        new_unified_dataset_name: new unified dataset name
+        exclude_artifacts: list of artifacts to exclude in import
+        include_additive_artifacts: list of artifacts to import only additively
+        include_destructive_artifacts: list of artifacts to import destructively
+        fail_if_not_present: flag to fail project if not already present in instance
+        asynchronous: flag to run function asynchronously
+        overwrite_existing: flag to overwrite existing project artifacts
+
+    Returns:
+        operation for project import api call
+    """
+    # check version compatibility for project movement
+    utils.version.enforce_after_or_equal(client=tamr_client, compare_version="2021.005.0")
+    logging.info("Tamr version has movement compatibility.")
+
+    # make project import api request
+    body = {
+        "projectArtifact": project_artifact_path,
+        "newProjectName": new_project_name,
+        "newUnifiedDatasetName": new_unified_dataset_name,
+        "excludeArtifacts": exclude_artifacts,
+        "includeAdditiveArtifacts": include_additive_artifacts,
+        "includeDestructiveArtifacts": include_destructive_artifacts,
+        "failIfNotPresent": fail_if_not_present,
+    }
+    if target_project:
+        if overwrite_existing:
+            logging.info(
+                f"importing existing project {target_project.name} "
+                f"with id {target_project.resource_id}."
+            )
+            response = tamr_client.post(
+                f"/v1/projects/{target_project.resource_id}:import", json=body
+            )
+        else:
+            logging.info("Unable to overwrite existing project; overwrite flag is off.")
+            raise KeyError("Unable to overwrite existing project; overwrite flag is off.")
+    else:
+        logging.info(f"Sending query to import new project, {new_project_name}.")
+        response = tamr_client.post("/v1/projects:import", json=body)
```

Comment:
There a specific API to import a new project, what is the reasoning for using the same import API as for existing projects?

Reply:
I'm not sure I understand the question. We are already using two separate APIs for import new project and for import into existing project.